### ISSUE TITLE
[Deprecation] remove `pallet::getter` usage from all pallets

### ISSUE THREAD

**Issue by muraca** [paritytech/polkadot-sdk#3326]
This issue tracks the removal of `pallet::getter` in all pallets, as per #223. 
This approach avoids a huge PR and lots of merge conflicts, a mistake I already made when I tried to perform this task last year.

Bridges:
- [ ] `pallet-bridge-grandpa`
- [ ] `pallet-bridge-messages`
- [ ] `pallet-bridge-relayers`
- [ ] `pallet-xcm-bridge-hub-router`
- [ ] `snowbridge-pallet-ethereum-client`
- [ ] `snowbridge-pallet-inbound-queue`
- [ ] `snowbridge-pallet-outbound-queue`
- [ ] `snowbridge-pallet-system`

Cumulus:
- [ ] `cumulus-pallet-aura-ext`
- [ ] `pallet-collator-selection`
- [ ] `cumulus-pallet-parachain-system`
- [ ] `pallet-parachain-template`
- [ ] `staging-parachain-info`

Polkadot:
- [ ] `polkadot-runtime-common`
- [ ] `polkadot-runtime-parachains`
- [ ] `pallet-xcm`
- [ ] `xcm-simulator-example`
- [ ] `xcm-simulator-fuzzer`

FRAME:
- [ ] `pallet-template`
- [ ] `pallet-alliance`
- [ ] `pallet-aura`
- [ ] `pallet-authority-discovery`
- [ ] `pallet-babe`
- [ ] `pallet-balances`
- [ ] `pallet-beefy`
- [ ] `pallet-beefy-mmr`
- [ ] `pallet-bounties`
- [ ] `pallet-child-bounties`
- [ ] `pallet-collective`
- [ ] `pallet-contracts-mock-network`
- [ ] `pallet-democracy`
- [ ] `pallet-election-provider-multi-phase`
- [ ] `pallet-elections-phragmen`
- [ ] `pallet-example-basic`
- [ ] `pallet-example-kitchensink`
- [ ] `pallet-example-offchain-worker`
- [ ] `pallet-fast-unstake`
- [ ] `pallet-grandpa`
- [ ] `pallet-identity`
- [ ] `pallet-im-online`
- [ ] `pallet-insecure-randomness-collective-flip`
- [ ] `pallet-membership`
- [ ] `pallet-mmr`
- [ ] `pallet-nft-fractionalization`
- [ ] `pallet-node-authorization`
- [ ] `pallet-offences`
- [ ] `pallet-proxy` 
- [ ] `pallet-recovery` 
- [ ] `pallet-sassafras` 
- [ ] `pallet-scored-pool` 
- [ ] `pallet-session` 
- [ ] `pallet-staking` 
- [ ] `pallet-state-trie-migration` 
- [ ] `pallet-sudo`
- [ ] `frame-support`
- [ ] `frame-system`
- [ ] `pallet-timestamp`
- [ ] `pallet-tips`
- [ ] `pallet-transaction-payment`
- [ ] `pallet-transaction-storage`
- [ ] `pallet-treasury`
- [ ] `pallet-vesting`
- [ ] `substrate-test-runtime` 

cc @kianenigma @shawntabrizi

**Comment by k-gunjan** [comment]
I want to work on pallet-aura

**Comment by PolkadotDom** [comment]
@muraca Did we come to consensus on what to do about storage values that lose visibility due to being pub(super), etc? Are we standing behind value::<T, I>::get() usage so staunchly that we should just mark them pub? Or should we implement getters such that they remain read-only?

**Comment by muraca** [comment]
In the past PRs I have done I have marked all the storage items as pub, however take my words with a grain of salt as I'm just a mere open source contributor, I don't want any responsibility on this 😄

**Comment by PolkadotDom** [comment]
Hahah sounds good! I'm going to try the 'implement a replacement getter' approach and we'll see if I get yelled at. I feel data authority is important enough for this to be the route.

**Comment by PolkadotDom** [comment]
#4417 for mock network by the way @muraca

**Comment by Aideepakchaudhary** [comment]
I want to work on Pallet-tips

**Comment by Aideepakchaudhary** [comment]
@muraca while removing the getter from pallet-babe, In test case I'm getting this issue:
`2024-06-28T12:02:12.641334Z ERROR runtime::timestamp: pallet_timestamp::UnixTime::now is called at genesis, invalid value returned: 0    
`

Here is the branch which I'm, working on: https://github.com/paritytech/polkadot-sdk/compare/master...openguild-labs:polkadot-sdk:deepak-remove-getter-from-babe-pallet?expand=1

Can you help me to resolve this issue?

**Comment by PolkadotDom** [comment]
> I want to work on Pallet-tips

Go for it! And to catch you up, basically all storage values are to be changed to pub (from pub(super) lets say) and all removes getters are to be reimplemented explicitly such that the public API remains the same. This is according to #223 and an example PR is #4840

**Comment by Zebedeusz** [comment]
I'm taking on `pallet-offences`

**Comment by Zebedeusz** [comment]
I'm starting to work on `pallet-staking`

**Comment by PolkadotDom** [comment]
@ggwpez It seems that upkeep for this issue has been abandoned or backgrounded as there have been a few open PRs for a while now that aren't tracked on the list, which will likely lead to duplicate work.

I could handle the tracking if you'd like, either through a new issue or getting edit access to this issue (not sure if that's feasible on github). Or if there's another point person we could assign whose more active, that would work as well.

**Comment by muraca** [comment]
@PolkadotDom as I previously stated, I am not a Parity employee nor affiliate in any form, and I took this task at a moment in which my free time and responsibilities allowed me to be much more present on Polkadot SDK. 
I'd be happy to hand this over to you, but as you stated, I don't think it's possible for you to edit this issue if you are not the author or part of the organization. 
I think the best option for you would be to create a new issue and close this one. In alternative, a Parity employee can take responsibility for this migration and keep tracking in this issue.

**Comment by PolkadotDom** [comment]
@muraca If still interested but just low on time, I could bundle up all the updates so it's quick for you to make the updates to the tracking list? Probably the easiest route forward.

**Comment by ggwpez** [comment]
>@ggwpez It seems that upkeep for this issue has been abandoned or backgrounded as there have been a few open PRs for a while now that aren't tracked on the list, which will likely lead to duplicate work.

Yes true. We kind of DDos'ed ourselves a bit here. The issue is also that these things are nice for newcomers but we dont have much capacity currently to review them.

Either way thanks for your help. Trying to review a few now 🤓

**Comment by PolkadotDom** [comment]
Sorry guys I wasn't meaning to point fingers or be critical of the pace. Was just a bit worried about the tracking list staying updated so newcomers didn't get confused. I appreciate everyones work 🙏
